Libraries/dotNetRDF/Writing/GraphMLWriter.cs

```diff
@@ -0,0 +1,282 @@
+        {
+            this.Save(store, output, false);
+        }
+
+        /// <summary>
+        /// Saves a triple store to a text writer in GraphML format
+        /// </summary>
+        /// <param name="store">The source triple store</param>
+        /// <param name="output">The target text writer</param>
+        /// <param name="leaveOpen">Boolean flag indicating if the output writer should be left open by the writer when it completes</param>
+        public void Save(ITripleStore store, TextWriter output, bool leaveOpen)
+        {
+            using (var writer = XmlWriter.Create(output, new XmlWriterSettings { CloseOutput = !leaveOpen, OmitXmlDeclaration = true }))
+            {
+                this.Save(store, writer);
+            }
+        }
+
+        /// <summary>
+        /// Saves a triple store to an XML writer in GraphML format
+        /// </summary>
+        /// <param name="store">The source triple store</param>
+        /// <param name="output">The target XML writer</param>
+        public void Save(ITripleStore store, XmlWriter output)
+        {
+            GraphMLWriter.WriteGraphML(output, store);
+        }
+
+        private static void WriteGraphML(XmlWriter writer, ITripleStore store)
+        {
+            writer.WriteStartElement(GraphMLHelper.GraphML, GraphMLHelper.NS);
+
+            GraphMLWriter.WriteKey(writer, GraphMLHelper.EdgeLabel, GraphMLHelper.Edge);
+            GraphMLWriter.WriteKey(writer, GraphMLHelper.NodeLabel, GraphMLHelper.Node);
+
+            foreach (var graph in store.Graphs)
+            {
+                GraphMLWriter.WriteGraph(writer, graph);
+            }
+
+            writer.WriteEndElement();
+        }
+
+        private static void WriteGraph(XmlWriter writer, IGraph graph)
+        {
+            writer.WriteStartElement(GraphMLHelper.Graph);
+
+            // Named graphs
+            if (graph.BaseUri != null)
+            {
+                writer.WriteAttributeString(GraphMLHelper.Id, graph.BaseUri.AbsoluteUri);
+            }
+
+            // RDF is always a directed graph
+            writer.WriteAttributeString(GraphMLHelper.Edgedefault, GraphMLHelper.Directed);
+
+            // The next two iterations are disjoint
+            foreach (var literalNode in graph.Nodes.Where(n => n.NodeType != NodeType.Literal))
+            {
+                GraphMLWriter.WriteNode(writer, literalNode.GetHashCode().ToString(), literalNode.ToString());
+            }
+
+            // Literal node identifiers are the whole statement so they become distinct
+            foreach (var triple in graph.Triples.Where(t => t.Object.NodeType == NodeType.Literal))
+            {
+                GraphMLWriter.WriteNode(writer, triple.GetHashCode().ToString(), triple.Object.ToString());
+            }
+
+            foreach (var triple in graph.Triples)
+            {
+                GraphMLWriter.WriteEdge(writer, triple);
+            }
+
```

Comment:
Would it be possible to merge the above two loops into a single one that writes the literal node if the object is a literal and then writes the edge? AFAIK GraphML doesn't care about the ordering of nodes/edges in the file.